scene/animation/tween.cpp

```diff
@@ -375,264 +375,10 @@ Variant Tween::interpolate_variant(Variant p_initial_val, Variant p_delta_val, f
 	ERR_FAIL_INDEX_V(p_trans, TransitionType::TRANS_MAX, Variant());
 	ERR_FAIL_INDEX_V(p_ease, EaseType::EASE_MAX, Variant());
 
-// Helper macro to run equation on sub-elements of the value (e.g. x and y of Vector2).
-#define APPLY_EQUATION(element) \
-	r.element = run_equation(p_trans, p_ease, p_time, i.element, d.element, p_duration);
-
-	switch (p_initial_val.get_type()) {
-		case Variant::BOOL: {
-			return (run_equation(p_trans, p_ease, p_time, p_initial_val, p_delta_val, p_duration)) >= 0.5;
```

Comment:
Ah, so nice pick, thanks.

Reply:
But I found this line in `Variant::interpolate()`
```
		default: {
			r_dst = c < 0.5 ? a : b;
		}
```

Reply:
I don't know what you replaced it with, but it works differently.

You can check with this code:
```GDScript
extends Sprite2D

func _ready() -> void:
	var tween = create_tween().set_loops()
	tween.tween_property(self, "visible", false, 1.0)
	tween.tween_property(self, "visible", true, 1.0)

func _notification(what: int) -> void:
	if what == NOTIFICATION_VISIBILITY_CHANGED:
		print(Time.get_ticks_msec())
```
Before it would print at half seconds, with your changes it prints at whole seconds.